tests/optimizers_tests/linear_model.py

```diff
@@ -0,0 +1,66 @@
+import numpy as np
+from chainer import cuda, Variable, FunctionSet
+from chainer.functions import Linear, softmax_cross_entropy, accuracy
+
+cuda.init()
+
+class LinearModel(object):
+    UNIT_NUM = 10
+    BATCH_SIZE = 32
+    EPOCH = 100
+
+    def __init__(self, optimizer):
+        self.model = FunctionSet(
+            l = Linear(LinearModel.UNIT_NUM, 2)
+        )
+        self.optimizer = optimizer
+        # true parameters
+        self.w      = np.random.uniform(-1, 1, (LinearModel.UNIT_NUM, 1)).astype(np.float32)
+        self.b      = np.random.uniform(-1, 1, (1, )).astype(np.float32)
+
+    def _train_linear_classifier(self, model, optimizer, gpu):
+        def _make_label(x):
+            a = (np.dot(x, self.w) + self.b).reshape((LinearModel.BATCH_SIZE, ))
```

Comment:
Use `self.BATCH_SIZE` instead of `LinearModel.BATCH_SIZE`, since class attributes can be overridden by child classes.